### PR TITLE
raw read/write: boolean support

### PR DESCRIPTION
**Pull request by rucub100**

### CODE REVIEW

File: src/main/java/de/hhu/bsinfo/dxutils/UnsafeMemory.java [hunk 1]
```diff
@@ -89,6 +89,26 @@ public static void copyBytes(final long p_address1, final long p_address2, final
...
+     *         Array to read the data into.
+     * @param p_arrayOffset
+     *         Start offset in array to start writing the booleans to.
+     * @param p_length
+     *         Number of booleans to read from specified start.
+     * @return Number of read elements.
+     */
+    public static int readBooleans(final long p_ptr, final boolean[] p_array, final int p_arrayOffset, final int p_length) {
+        MS_UNSAFE_HANDLER.getUnsafe().copyMemory(null, p_ptr, p_array,
+                UnsafeHandler.getArrayByteOffset() + p_arrayOffset, p_length);
```
Comment (krakowski) [inline src/main/java/de/hhu/bsinfo/dxutils/UnsafeMemory.java]: The base offset is not intended for boolean arrays. Please add a new method (`getArrayBooleanOffset`), which returns the appropriate value `sun.misc.Unsafe.ARRAY_BOOLEAN_BASE_OFFSET`.

File: src/main/java/de/hhu/bsinfo/dxutils/UnsafeMemory.java [hunk 2]
```diff
@@ -313,6 +344,26 @@ public static double readDouble(final long p_ptr) {
...
+     * @param p_array
+     *         Array with data to write.
+     * @param p_arrayOffset
+     *         Offset in array to start reading the data from.
+     * @param p_length
+     *         Number of elements to write.
+     * @return Number of written elements
+     */
+    public static int writeBooleans(final long p_ptr, final boolean[] p_array, final int p_arrayOffset, final int p_length) {
+        MS_UNSAFE_HANDLER.getUnsafe().copyMemory(p_array, UnsafeHandler.getArrayByteOffset() + p_arrayOffset, null,
```
Comment (krakowski) [inline src/main/java/de/hhu/bsinfo/dxutils/UnsafeMemory.java]: The base offset is not intended for boolean arrays. Please add a new method (`getArrayBooleanOffset`), which returns the appropriate value `sun.misc.Unsafe.ARRAY_BOOLEAN_BASE_OFFSET`.

**Review by krakowski — CHANGES_REQUESTED** [review]
Left some comment regarding array base offsets. The other parts look good.

File: src/main/java/de/hhu/bsinfo/dxutils/UnsafeMemory.java (hunk 1 above)
Comment (rucub100) [reply src/main/java/de/hhu/bsinfo/dxutils/UnsafeMemory.java]: Done

File: src/main/java/de/hhu/bsinfo/dxutils/UnsafeMemory.java (hunk 2 above)
Comment (rucub100) [reply src/main/java/de/hhu/bsinfo/dxutils/UnsafeMemory.java]: Done

**Review by krakowski — APPROVED** [review]

**Pull request merged**